qp_qiime2/__init__.py

```diff
@@ -76,234 +86,191 @@
+        # emperor procrustes_plot
+        # diversity pcoa_biplot
+        # diversity procrustes_analysis
+        # gneiss assign_ids
+        # gneiss ilr_phylogenetic
+        # gneiss ilr_phylogenetic_differential
+        # taxa filter_seqs
+        continue
 
-            qiime_cmd = QiitaCommand("%s [%s]" % (m.name, m.id), m.description,
-                                     call_qiime2, req_params, opt_params,
-                                     outputs_params, {'Default': {}},
-                                     analysis_only=True)
+    for pname, element in parameters.items():
+        tqt, predicate = get_qiime2_type_name_and_predicate(element)
+        # there is a new primitive and we should raise an error
+        if tqt not in PRIMITIVE_TYPES:
+            raise ValueError(
+                'There is a new type: %s, in %s %s (%s)' % (
+                    tqt, q2plugin.name, m.id, pname))
 
-            plugin.register_command(qiime_cmd)
+        # predicate are the options for each parameter, note that it
+        # can be a Choice/List or a Range (for Int/Floats). We ignore
+        # numeric because they are ranges and we don't support ranges
+        # in Qiita.
+        # Note, the correct way to retrieve the latter is:
+        # p.start, p.end, p.inclusive_start, p.inclusive_end
+        # but for simplicity we will only retrive the
+        data_type = PRIMITIVE_TYPES[tqt]
+        default = element.default
+        if (predicate is not None and PRIMITIVE_TYPES[tqt] not in (
+                                      'float', 'integer')):
+            vals = predicate.to_ast()['choices']
+            data_type = 'choice:%s' % dumps(vals)
+            default = vals[0]
 
-# 2. force adding extra commands
-pm = PluginManager()
+        qname = q2plugin.name
+        mid = m.id
+        # if we are in the diversity plugin, the method starts with
+        # alpha/beta, and the parameter is 'metric', we might want
+        # to replace the technical names for user friendly ones
+        value_pair = (mid, pname)
+        if qname == 'diversity' and value_pair in RENAME_COMMANDS:
+            # converting to list to the serialize doesn't complaint
+            vals = list(RENAME_COMMANDS[value_pair])
+            data_type = 'choice:%s' % dumps(vals)
+            default = vals[0]
 
-# 2.1 adding assign taxonomy
-q2plugin = pm.plugins['feature-classifier']
-m = q2plugin.methods['classify_sklearn']
-qname = q2plugin.name
-mid = m.id
+        # the diversity methods can have a choice param with no values
+        # so we need to fix so users can actually select things;
+        # however,we want to make sure that this is the only one,
+        # if not, raise an error
+        if data_type == 'choice' and default is None:
+            # these are valid optional parameters
+            valid_none_params = [
+                ('emperor', 'plot'),
+                ('gneiss', 'ilr_phylogenetic_ordination')]
```

Comment:
Finally, the other "important" change is that we add `('gneiss', 'ilr_phylogenetic_ordination')` to here as this is also a valid None value (optional)